.github/workflows/ci-cd.yml

```diff
@@ -106,32 +106,34 @@ jobs:
         run: |
           case "$OS" in
             ubuntu*)
-              echo "path-sep=/" >>$GITHUB_OUTPUT
-              echo "pip-cache=~/.cache/pip" >>$GITHUB_OUTPUT
+              echo "pip-cache=~/.cache/pip" >> $GITHUB_OUTPUT
               ;;
             macos*)
-              echo "path-sep=/" >>$GITHUB_OUTPUT
-              echo "pip-cache=~/Library/Caches/pip" >>$GITHUB_OUTPUT
+              echo "pip-cache=~/Library/Caches/pip" >> $GITHUB_OUTPUT
               ;;
             windows*)
-              echo "path-sep=\\" >>$GITHUB_OUTPUT
-              echo "pip-cache=~\\AppData\\Local\\pip\\Cache" >>$GITHUB_OUTPUT
+              echo "pip-cache=~\\AppData\\Local\\pip\\Cache" >> $GITHUB_OUTPUT
               ;;
           esac
+          echo "date=$(date +%Y%m%d 2> /dev/null || true)" >> $GITHUB_OUTPUT
         shell: bash
 
       - name: Cache PIP Packages
         uses: actions/cache@v3
         id: cache
         with:
           path: ${{ steps.os.outputs.pip-cache }}
-          key: ${{ matrix.os }}-pip-test-${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt', '**/constraints.txt') }}
+          key: ${{ matrix.os }}-pip-test-${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt', '**/constraints.txt') }}-${{ steps.os.outputs.date }}
```

Comment:
doesn't restore-keys automatically pick the latest?

Reply:
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key:~:text=If%20there%20are%20multiple%20partial%20matches%20for%20a%20restore%20key%2C%20the%20action%20returns%20the%20most%20recently%20created%20cache.

Reply:
This is to mitigate a different facet of the cache: if there is a cache hit on the primary key, the cache will not be saved, changes to the cached files will not propagate to the next run. With this, it will safe the cache once per day.

Alternative is to use [actions/cache/save](https://github.com/actions/cache/blob/main/save/README.md) and [actions/cache/restore](https://github.com/actions/cache/blob/main/restore/README.md).

Reply:
ah, makes sense. I think an explicit save might be better, because it would take up way less space. If a workflow is ran every day, the cache will be populated every day, but the older caches will never be used. While with `cache/save` there would be only one cache, that gets refreshed every day.

https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache:~:text=the%20total%20size%20of%20all%20caches%20in%20a%20repository%20is%20limited%20to%2010%20GB.

If I read this right, these_ together mean that with the current solution other caches of the user might get evicted, because of the unused 6 days worth of pip caches.

Reply:
I am not using the date in the key for the composite action, only for my own CI